collector/exporter/otelarrowexporter/internal/arrow/common_test.go

```diff
@@ -158,12 +161,14 @@ func (ctc *commonTestCase) repeatedNewStream(nc func() testChannel) func(context
 		str := ctc.newMockStream(ctx)
 		str.sendCall.AnyTimes().DoAndReturn(h.onSend(ctx))
 		str.recvCall.AnyTimes().DoAndReturn(h.onRecv(ctx))
+		str.closeSendCall.AnyTimes().DoAndReturn(h.onCloseSend())
 		return str.anyStreamClient, nil
 	}
 }
 
 // healthyTestChannel accepts the connection and returns an OK status immediately.
 type healthyTestChannel struct {
+	lock sync.Mutex
```

Comment:
```suggestion
    // This lock is needed to avoid a race because CloseSend() now closes the sent channel, 
    // which some former tests were doing manually. Now CloseSend() is always called and
    // the closed channel serves to assist with some tests.
	lock sync.Mutex
```